File: Problem5.java
package leetcode;

import java.util.Scanner;

public class Problem5 {
    
    public String longestPalindrome(String s) {
        String maxStr = "";
        if(s == null || s.length() == 0) {
            return "";
        }
        if(s.length() == 1) {
            return s;
        }
        for(int i=0;i<s.length();i++) {
            maxStr = maxStr(i, i, s, maxStr);
            maxStr = maxStr(i, i+1, s, maxStr);
        }
        return maxStr;
    }
    
    private String maxStr(int back, int front, String s, String maxStr) {
        while(back >= 0 && front < s.length() && s.charAt(back) == s.charAt(front)) {
            back--;
            front++;
        }
        if(front-back > maxStr.length()) {
            System.out.println(front);
            System.out.println(back);
            return s.substring(back+1, front);
        }
        return maxStr;
    }
    

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int t = scanner.nextInt();
        for (int a0 = 0; a0 < t; a0++) {
            String line = scanner.next();
            Problem5 problem = new Problem5();
            System.out.println(problem.longestPalindrome(line));
        }
        scanner.close();
    }
    
}

File: problem5.go
package main

import "fmt"

func longestPalindrome(s string) string {
    if len(s) == 0 {
        return ""
    }
    if len(s) == 1 {
        return s
    }
    var maxstr = ""
    var i int
    for i=0; i<len(s); i++ {
        maxstr = maxStr(i, i, s, maxstr)
        maxstr = maxStr(i, i+1, s, maxstr)
    }
    return maxstr
}


func maxStr(back int, front int, s string, maxstr string) string {
    for back>=0 && front<len(s) && s[back] == s[front] {
        back--
        front++
    }
    
    if (front-back) > len(maxstr) {
        return s[back+1:front]
    }
    
    return maxstr
}


func main() {
    var maxstr = longestPalindrome("babad")
    fmt.Println(maxstr)
    maxstr = longestPalindrome("cbbbd")
    fmt.Println(maxstr)
    maxstr = longestPalindrome("cbbd")
    fmt.Println(maxstr)
}
